src/main.rs

```diff
@@ -576,9 +594,22 @@ fn db_connect(creds_file: PathBuf) -> Pool {
 async fn listen(listen_address: String, listen_port: u16) -> std::io::Result<()> {
     log::info!("start listening on port: {}", listen_port);
 
+    let max_file_descriptors = open_file_limit().await;
     let manifest_url = "/v2/{name}/manifests/{reference}";
     let blob_url = "/v2/{name}/blobs/{reference}";
 
+    tokio::spawn(async move {
+        loop {
+            tokio::time::sleep(FD_CHECK_INTERVAL).await;
+            let current_fds = get_open_fds_count().await;
+            log::info!("number of open filedescriptors: {current_fds}");
+            if current_fds > max_file_descriptors {
+                eprintln!("CRITICAL: Too many open file descriptors: {}", current_fds);
+                std::process::exit(1);
+            }
+        }
+    });
```

Comment:
I'm not a hugefan of this approach. Instead of having a periodical "watch dog", perhaps we could instead have a check that's triggered at the point where file descriptors are consumed, or potentially, we could make it so that when a file descriptor is attempted to be consumed, but too many are open, the respective event fails.

This would require triaging where the actual issue occurs, but then we could imagine that greedy tasks that take up too many resources would die, rather than having the ability to bring down wharfix.

Also, in extension of talk about secure design, the current timer approach seems more susceptible to denial of service.